workflow/parser.py

```diff
@@ -0,0 +1,67 @@
+import os
+import subprocess
+
+
+class Parser:
+    def __init__(self, params):
+        self.params = params
+
+    def parse(self):
+
+        '''
+        - parse input config file 
+        - build TPS call 
+        - call TPS 
+        - gather outputs 
+        - return:
+            - list of outputs 
+            - output_folder
+            - out_label
+
+      
+        '''
+        # # grab TPS self.params 
+        build = []
+        OUT_FOLDER = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+        OUT_LABEL = ""
+
+        # add TPS run call as first arg
+        build.extend(["bash", "./scripts/run"])
+
+        # grab self.params from config
+        req = self.params[1]["TPS"][0]["required"]
+        op = self.params[1]["TPS"][1]["optional"]
+
+        # filter given args
+        filtered = {**req, **op}
+        args = {key:val for key, val in filtered.items() 
+                if filtered[key] != ""}
```

Comment:
Is this logic saying that we skip optional arguments for which an empty value was provided?  I'd find that a little confusing as a user.  Should we tell users to delete or comment out those arguments instead?

Reply:
I've undated this logic with to filter required and optional arguments with "None" flag and no value arguments as a list in the config file